### DISCUSSION TITLE
Pressing "Tab" key to accept suggestions does not work in Visual Studio 2022 17.6.0 Preview 1.0

### DISCUSSION THREAD

**Discussion by Pr0ject414** [community/community#49212] [Copilot Conversations] [answered]
### Select Topic Area

Bug

### Body

Pressing "Tab" key to accept suggestions no longer works in Visual Studio 2022. 

I am forced to use the "->" right arrow key to accept the suggestion from copilot. I have updated several times and still no longer have the ability to hit the "Tab" key to accept suggestions as I previously was able to do. 

I am using Copilot Version 1.77.0.0, which was installed on 03/02/2023. 
I am on Visual Studio 2022 17.6.0 Preview 1.0

Was this functionality removed and replaced? I am unable to find the keyboard bindings in the Tools->Options->Environment->Keyboard when searching for Copilot Commands.

**Comment by MaxWilson** [comment]
Ironically, your bug sounds like my workaround--I NEED Copilot to stop inserting code just because I wanted an indentation (whitespace is significant in F#). I wish I could repro your bug because it would make my life a lot easier.

**Comment by TheYearIsTwentyGX** [comment]
Also having this issue. Pressing tab usually just ends up using Intellisense completions instead. Very annoying thinking that I'm going to insert several lines of code that have been suggested, but then I end up just putting "#region" or ""

**Comment by k-nero** [comment]
I'm not sure about your situation but this work for me.
Try to disable the option "Automatically indent when i type a tab" in VS2022 by Tools -> Options -> Text Editor -> C/C++ -> Code style -> Formatting -> General

**Reply by TristofCuber** [reply]
Thanks, this solution worked for me too!

**Reply by chrkspln** [reply]
thanks a lot, it finally works as needed

**Reply by hegworks** [reply]
You are a life-saver. Thank you so much.

**Comment by Robert-Tortorelli** [comment]
In Visual Studio, Tools/ Options…/ IntelliCode/ Advanced, enabling the setting 'Apply whole line completions on right arrow' _prevents the Tab key from ‘accepting’ a GitHub Copilot suggestion_. 
Prior to disabling this setting, the right arrow, but not Tab, accepted suggestions. However I think these suggestions were coming from IntelliCode, not GitHub Copilot.
'Apply whole line completions on right arrow' was enabled by default on my installation of Visual Studio.
Microsoft Visual Studio Community 2022
Version 17.7.6
VisualStudio.17.Release/17.7.6+34221.43
Microsoft .NET Framework
Version 4.8.09032

Installed Version: Community

Visual C++ 2022   00482-90000-00000-AA530
Microsoft Visual C++ 2022

Azure App Service Tools v3.0.0   17.7.274.37181
Azure App Service Tools v3.0.0

C# Tools   4.7.0-3.23517.17+9d4cc0304792762c34f41cbbec67a1fcd2dc764f
C# components used in the IDE. Depending on your project type and settings, a different version of the compiler may be used.

GitHub Copilot   1.133.0.0 (v1.133.0.0@a12ebe1a4)
GitHub Copilot is an AI pair programmer that helps you write code faster and with less work.

GitHub Copilot Agent   1.133.0

Microsoft JVM Debugger   1.0
Provides support for connecting the Visual Studio debugger to JDWP compatible Java Virtual Machines

NuGet Package Manager   6.7.0
NuGet Package Manager in Visual Studio. For more information about NuGet, visit https://docs.nuget.org/

Visual Basic Tools   4.7.0-3.23517.17+9d4cc0304792762c34f41cbbec67a1fcd2dc764f
Visual Basic components used in the IDE. Depending on your project type and settings, a different version of the compiler may be used.

Visual Studio IntelliCode   2.2
AI-assisted development for Visual Studio.

Visual Studio Spell Check Everywhere   VSSpellCheckEverywhere
An extension that enables spell checking within any Visual Studio file editor or tool window that uses WPF text boxes.
https://GitHub.com/EWSoftware/VSSpellChecker

Visual Studio Spell Checker   VSSpellChecker
An editor extension that checks the spelling of comments, strings, and plain text as you type or interactively with tool windows.
https://GitHub.com/EWSoftware/VSSpellChecker

**Reply by marekmalczewski** [reply]
You have saved my life! Thank you.

**Comment by hmackenberg** [comment]
For me, enabling the setting 'Apply whole line completions on right arrow' in Tools/ Options…/ IntelliCode/ Advanced worked: it allows accepting suggestions using the right arrow key

**Reply by Robert-Tortorelli** [reply]
True. But the issue I have is that, as you indicate, this is an **IntelliCode** option, not a **Copilot** option. Enabling an IntelliCode option should not affect Copilot, especially as _Copilot's own GitHub documentation specifies using Tab to accept Copilot suggestions_. See Step 3 under https://docs.github.com/en/copilot/using-github-copilot/getting-started-with-github-copilot#seeing-your-first-suggestion-1

This IntelliCode option was enabled when I installed Microsoft Visual Studio Community 2022. However, clicking 'Use defaults' on this options page does disable 'Apply whole line completions on right arrow' and restore the use of Tab to accept Copilot suggestions.

**Reply by ayim** [reply, marked as answer]
Hello from Visual Studio team, changing the IntelliCode setting DOES in fact change the Copilot Completions behaviour.
This occurs because Copilot Completions uses the IntelliCode component to display and commit predictions.

![image](https://github.com/community/community/assets/10951869/5d40bc03-9969-43ee-9f8d-9de6ae88ddd6)
![image](https://github.com/community/community/assets/10951869/3c128627-9d08-4c53-8f55-750ec43eb85d)

**Reply by GeorgeSeeger** [reply]
This was the right fix for me, thank you

**Comment by aureole82** [comment]
This doesn't work in Visual Studio 2026 v18.3.0:
<img width="873" height="382" alt="image" src="https://github.com/user-attachments/assets/f8f3afbf-d570-46b5-b34c-70a0e99aabdd" />
It still want's the `Tab` key:
<img width="746" height="200" alt="image" src="https://github.com/user-attachments/assets/012e803b-c686-42d9-bab1-bcad7d69c199" />
And hitting `Tab`:
<img width="688" height="51" alt="image" src="https://github.com/user-attachments/assets/1ca32c64-a231-4a16-b7ca-396de98f2c57" />

_2026-04-17_: Just updated to Visual Studio 18.5.0. **It's still the same!**

**Reply by WillBellJr** [reply]
I'm on Community Insiders [11605.296]
And I'm sitting here simply trying to **accept** the ending text of a Xaml (Avalonia UI) comment and pressing Tab as suggested totally deletes the suggested text at the end of the comment??

I've pressed Tab to accept, right-arrow, attempted pressing 6 (>) on the number pad, no key I type will fill in the suggested text for me. 😪

I can't say I've changed any of the settings from their defaults.
